x-pack/plugins/security_solution/cypress/screens/alerts.ts

```diff
@@ -41,6 +41,8 @@ export const ACKNOWLEDGED_ALERTS_FILTER_BTN = '[data-test-subj="acknowledgedAler
 
 export const LOADING_ALERTS_PANEL = '[data-test-subj="loading-alerts-panel"]';
 
+export const LOADING_SPINNER = '.SpinnerFlexItem-zdczxa-0';
```

Comment:
Class names generated by `styled-components ` are not stable (can change with time) and should not be used as selectors for test purposes.
Can we add `data-test-subj` attribute to `SpinnerFlexItem ` and use it to target the loading state instead?

Reply:
@xcrzx I'm not 100% sure how to do it for that piece of code, If you point me out where I should add it I'm more than happy to do it. Thanks :)

Reply:
Hmm, it looks like there is already a `test-subj`, but it is not applied to DOM due to a typo.
Here is the fix:
```diff
diff --git a/x-pack/plugins/timelines/public/components/t_grid/footer/index.tsx b/x-pack/plugins/timelines/public/components/t_grid/footer/index.tsx
index c9e50d74961..a4bc2bf9133 100644
--- a/x-pack/plugins/timelines/public/components/t_grid/footer/index.tsx
+++ b/x-pack/plugins/timelines/public/components/t_grid/footer/index.tsx
@@ -287,7 +287,7 @@ export const FooterComponent = ({
     return (
       <LoadingPanelContainer>
         <LoadingPanel
-          data-test-subj="LoadingPanelTimeline"
+          dataTestSubj="LoadingPanelTimeline"
           height="35px"
           showBorder={false}
           text={loadingText}

```

And after that you can use `export const LOADING_SPINNER = '[data-test-subj="LoadingPanelTimeline"]';`